result.h

```diff
@@ -844,6 +844,20 @@ struct Result {
         std::terminate();
     }
 
+    template<typename F = E>
+    typename std::enable_if<
+        !std::is_same<F, void>::value,
+        F
+    >::type
```

Comment:
Right, but I think it will be a little bit painful to handle.

Currently, you might want to try defining a unit type like `struct Unit { }` and use it as a void error type, what do you think ?